app/lib/lapin/range.rb

```diff
@@ -4,10 +4,10 @@ module Lapin
   module Range
     class << self
       def reduce_range_to_delay(motif, date_range)
-        return nil if date_range.end < (Time.zone.now + motif.min_booking_delay.seconds)
+        return nil if !motif.booking_delay_range.cover?(date_range.end) && !motif.booking_delay_range.cover?(date_range.begin)
```

Comment:
C'est ici le point clef pour la correction.

Reply:
```suggestion
        return nil unless motif.booking_delay_range.overlaps?(date_range)
```
À vue de nez, c’est strictement équivalent, [l’implémentation de `Range#overlaps?` fait ça](https://github.com/rails/rails/blob/6-1-stable/activesupport/lib/active_support/core_ext/range/overlaps.rb).